source/features/mark-unread.js

```diff
@@ -367,8 +367,10 @@ export default async function () {
 				delegate('.btn-link.delete-note', 'click', markNotificationRead),
 				delegate('.js-mark-all-read', 'click', markAllNotificationsRead),
 				delegate('.js-delete-notification button', 'click', updateUnreadIndicator),
-				delegate('form[action="/notifications/mark"] button', 'click', () => {
-					storage.set([]);
+				delegate('form[action="/notifications/mark"] button', 'click', e => {
+					const group = e.target.closest('.boxed-group');
```

Comment:
Yeah thought of it, but you won't be needing it, as `delegateTarget` is just `e.target.closest` applied with the given selector.

Because we need to perform `closest` anyway, and also we don't need the element on which the event was generated (delegated target).